src/InputFilter.php

```diff
@@ -1006,7 +1006,7 @@ private function cleanPath($source)
 			return preg_replace('~/+~', '/', $source);
 		}
 
-		$windowsPattern = '/^([A-Z]:\\\\)?[A-Za-z0-9_\/-]+[A-Za-z0-9_\.-]*(\\\\+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/';
+		$windowsPattern = '/^([A-Z])?[A-Za-z0-9_\/:\\\\-]+[A-Za-z0-9_\.-]*(\\\\+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/';
```

Comment:
Will `/^([A-Za-z]:\\\\)?[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*(\\\\+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/` not be more correct? AFAIK, Windows drives are only 1 letter...

Reply:
Thats not the problem. The problem is that ":\\\\" part. ;)

![image](https://user-images.githubusercontent.com/2596554/104971263-839b1780-59ee-11eb-93b3-84e737400aa1.png)
vs
![image](https://user-images.githubusercontent.com/2596554/104971276-8eee4300-59ee-11eb-87d9-4dd4266e3a2d.png)

Reply:
I tried with these testcases:
```
			'windows path without extension' => [
				'path',
				'C:\Documents\Newsletters',
				'C:\Documents\Newsletters',
				'From generic cases'
			],
			'windows path with lower case drive letter' => [
				'path',
				'c:\Documents\Newsletters',
				'c:\Documents\Newsletters',
				'From generic cases'
			],
			'windows path with two drive letters' => [
				'path',
				'CC:\Documents\Newsletters',
				'',
				'From generic cases'
			],
			'windows path without drive letter' => [
				'path',
				'Documents\Newsletters',
				'Documents\Newsletters',
				'From generic cases'
			],
```

Only 'windows path with lower case drive letter' is failing, so the correct RegEx should be
'/^([A-Za-z]:\\\\)?[A-Za-z0-9_\/-]+[A-Za-z0-9_\.-]*(\\\\+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/'

Reply:
> The problem is that :\\\\ part. ;)

Well, escaping is not always portable from one environment (PHP string) to another (RegEx Tester), so I'll rather rely on the tests.

Reply:
BTW:
> The problem is that :\\\\ part. ;)
No, the problem is the closing bracket. You have `([A-Za-z:\\\\])?` instead of `([A-Za-z]:\\\\)?`

Reply:
hmm it seems also with the closing bracket it does not match?

![image](https://user-images.githubusercontent.com/2596554/104973744-f956b180-59f5-11eb-9af1-8abfd171be3f.png)

Or I'm still doing something wrong there?

Reply:
I have just commited the suggested changes, I'm not sure whether that is correct as it still fails on https://regexr.com/